core/modules/gtpu_encap.cc

```diff
@@ -193,21 +193,21 @@ CommandResponse GtpuEncap::Init(const bess::pb::GtpuEncapArg &arg) {
 
   using AccessMode = bess::metadata::Attribute::AccessMode;
   pdu_type_attr = AddMetadataAttr("action", sizeof(uint8_t), AccessMode::kRead);
-  DLOG(INFO) << "tout_sip_attr: " << tout_sip_attr << std::endl;
+  DLOG(INFO) << "tout_sip_attr: " << tout_sip_attr;
   tout_sip_attr = AddMetadataAttr("tunnel_out_src_ip4addr", sizeof(uint32_t),
                                   AccessMode::kRead);
-  DLOG(INFO) << "tout_sip_attr: " << tout_sip_attr << std::endl;
+  DLOG(INFO) << "tout_sip_attr: " << tout_sip_attr;
   tout_dip_attr = AddMetadataAttr("tunnel_out_dst_ip4addr", sizeof(uint32_t),
                                   AccessMode::kRead);
-  DLOG(INFO) << "tout_dip_attr: " << tout_dip_attr << std::endl;
+  DLOG(INFO) << "tout_dip_attr: " << tout_dip_attr;
   tout_teid =
       AddMetadataAttr("tunnel_out_teid", sizeof(uint32_t), AccessMode::kRead);
-  DLOG(INFO) << "tout_teid: " << tout_teid << std::endl;
+  DLOG(INFO) << "tout_teid: " << tout_teid;
   tout_uport = AddMetadataAttr("tunnel_out_udp_port", sizeof(uint16_t),
                                AccessMode::kRead);
-  DLOG(INFO) << "tout_uport: " << tout_uport << std::endl;
+  DLOG(INFO) << "tout_uport: " << tout_uport;
   qfi_attr = AddMetadataAttr("qfi", sizeof(uint8_t), AccessMode::kRead);
-  DLOG(INFO) << "qfi_attr: " << qfi_attr << std::endl;
+  DLOG(INFO) << "qfi_attr: " << qfi_attr;
```

Comment:
std::endl will immediately push prints (std:cout may not without it), could you confirm that without that this logging system will also immediately push prints immediately

Reply:
Yes, it does NOT need std::endl to print. I have tested it and I typically use it that way. Here is an example that I am currently using
```cpp
      LOG(INFO) << "type:" << +gtpuType
                << ", dstIp:" << ToIpv4Address(static_cast<be32_t>(dst))
                << ", seqNumber:" << seqNumber << ", latency:["
                << m_latency[dst].m_min << ", " << m_latency[dst].m_mean << ", "
                << m_latency[dst].m_max << "]";
```
and here is the output:
```bash
I1028 12:14:05.391299 106395 gtpu_echo_processing.cc:118] type:2, dstIp:10.10.10.10, seqNumber:0, latency:[7548, 7548, 7548]
I1028 12:14:05.391343 106395 gtpu_echo_processing.cc:118] type:2, dstIp:172.17.1.10, seqNumber:0, latency:[30106, 30106, 30106]
I1028 12:14:05.391348 106395 gtpu_echo_processing.cc:118] type:2, dstIp:192.168.0.55, seqNumber:0, latency:[32502, 32502, 32502]
I1028 12:14:12.566440 106395 gtpu_echo_processing.cc:118] type:2, dstIp:10.10.10.10, seqNumber:1, latency:[7548, 10132, 12716]
I1028 12:14:12.566478 106395 gtpu_echo_processing.cc:118] type:2, dstIp:172.17.1.10, seqNumber:1, latency:[30106, 40387, 50668]
I1028 12:14:12.566481 106395 gtpu_echo_processing.cc:118] type:2, dstIp:192.168.0.55, seqNumber:1, latency:[32502, 42350, 52199]
I1028 12:14:19.741501 106395 gtpu_echo_processing.cc:118] type:2, dstIp:10.10.10.10, seqNumber:2, latency:[7524, 9263, 12716]
I1028 12:14:19.741536 106395 gtpu_echo_processing.cc:118] type:2, dstIp:172.17.1.10, seqNumber:2, latency:[26379, 35718, 50668]
I1028 12:14:19.741539 106395 gtpu_echo_processing.cc:118] type:2, dstIp:192.168.0.55, seqNumber:2, latency:[27758, 37486, 52199]
```